protocols/mempool-overview.md

```diff
@@ -100,43 +100,45 @@ of the application, which is part of this same procedure to commit a block.
 
 ## Formalization Attempt
 
-Let `committed(tx, h)` return true iff the transaction `tx` is committed by
-height `h` of the blockchain.
-This means that `tx` was included in a block `B_k` of transactions with height `k <= h`.
-
-> Notice that `committed(tx, h) => committed(tx, h')` for all `h' > h`.
-
-Lets `valid(tx, h)` to return true iff the transaction `tx` is valid at height
-`h` of the blockchain.
-
-> The height of the blockchain reflects the state of the application.
-> Some transactions can be valid at a given state (height), but no longer valid
-> in another state (height).
-
-We need to assume for the `valid` predicate a property similar to the one we
-have for the `committed` predicate.
-That is, that from a given height the state of the transaction is not changed
-anymore.
-While this is trivial for the `committed` predicate, we need to assume that
-behavior for the `valid` predicate.
-
-So, we need to assume that there is a height `h*` from which a transaction `tx`
-becomes invalid, and this state does not change for higher heights.
-That is, we assume that there is a height `h*` so that for all `h >= h* =>
-valid(tx, h) == FALSE`.
-
-This property is realistic as applications are expected to implement some form
-of replay protection.
-This means that once a transaction `tx` is committed, the application should
-reject processing the same transaction `tx` again.
-This is achieved by rejecting `tx` as invalid from the moment that `tx` is
-committed and executed by the application.
-So, a replay protection procedure can be define as:
-`committed(tx, h) => h' > h \and !valid(tx, h')`
-
-Given these definitions, the main property of the mempool is the following:
-
- - If a transaction `tx` is added to the local mempool of a correct node, then
-   eventually we have a height `h*` for which either:
-   - `committed(tx, h*)` is true
-   - or `!valid(tx, h*)` and `h > h* => !valid(tx, h)`
+In what follows, we try to formalize the notion of mempool.
+To this end, we first provide a (brief) definition of what is a ledger, that is a replicated log of transactions.
+At some process $p$, we shall write $p.var$ the local variable $var$ at $p$.
+
+**Ledger.**
+We use the standard defintion of (BFT) SMR, where each process $p$ has a ledger, written $p.ledger$.
+At process $p$, the $i$-th entry of the ledger is denoted $p.ledger[i]$.
+This entry contains either a null value ($\bot$), or a set of transactions, aka., a block.
+The height of the ledger at $p$ is the index of the first null entry.
+Operation $submit(txs, i)$ attempts to write the set of transactions $txs$ to the $i$-th entry of the ledger.
```

Comment:
I would rather use `propose` here, using consensus nomenclature.

Reply:
I am using the term `submit` because the operation is part of the interface of state-machine replication (SMR). Under the hood, it is implemented as `ledger[i].propose(txs)`.